---
title: 1.8 pyav
createTime: 2025/04/20 14:35:32
permalink: /experience/j649g5o1/
---

## 1.8.1 The `pyav` plugin is not installed.

完整报错：

```bash
Traceback (most recent call last):
  File "/mnt/data/home/xxxx/WHAM/demo.py", line 226, in <module>
    run(cfg,
  File "/mnt/data/home/xxxx/WHAM/demo.py", line 180, in run
    run_vis_on_demo(cfg, video, results, output_pth, network.smpl, vis_global=run_global)
  File "/mnt/data/home/xxxx/WHAM/lib/vis/run_vis.py", line 171, in run_vis_on_demo
    writer = get_writer(osp.join(output_pth, 'output_global.mp4'), fps=30, crf=23)
  File "/mnt/data/home/xxxx/WHAM/lib/utils/video_io_utils.py", line 86, in get_writer
    writer = iio.imopen(video_path, "w", plugin="pyav")
  File "/mnt/data/home/xxxx/miniforge3/envs/wham/lib/python3.9/site-packages/imageio/core/imopen.py", line 156, in imopen
    raise err_type(err_msg) from err_from
ImportError: The `pyav` plugin is not installed. Use `pip install imageio[pyav]` to install it.
```

在使用 imageio 的时候：

```python
import imageio.v3 as iio
iio.imopen(video_path, "w", plugin="pyav") 
```

解决：

```bash
pip install av
```

